### DISCUSSION TITLE
[2주차] HTTP 1.0과 1.1의 차이가 무엇일까요?

### DISCUSSION THREAD

**Discussion by punchdrunkard** [prgrms-web-devcourse/We-are-the-MASTERS-OF-HTTP#26] [Interview Question]
HTTP 1.0과 1.1의 차이가 무엇일까요? (커넥션 관리의 관점에서 말씀해주시면 됩니다.)

**Comment by yunjjeongjo** [comment]
### 1. 지속성(Connection: Keep-alive 속성)
HTTP 1.0에서 요청하고 수신할 때마다 새로운 TCP 세션을 맺어야 하지만,
HTTP 1.1부터는 TCP 세션을 한 번만 맺으면 여러 개의 요청을 보내고 응답을 수신할 수 있습니다.

결과적으로 TCP 세션을 처리하는 비용을 줄이고 응답속도를 개선할 수 있게 됩니다.

### 2. 파이프라이닝
HTTP 1.0 환경에서는 요청에 대한 응답이 와야 다음 응답을 보낼 수 있지만,
HTTP 1.1에서는 요청을 병렬로 처리할 수 있는 파이프라이닝(Pipelining) 기능을 지원합니다.

결과적으로 HTTP 1.1에서는 여러 개의 요청을 처리하는 응답 속도가 훨씬 빨라지게 됩니다.
